### PR TITLE
Finalize config management reference documentation

### PR DESCRIPTION
**Pull request by didier-wenzek**
## Proposed changes

Update the reference documentation accordingly to https://github.com/thin-edge/thin-edge.io/pull/1145:
* Optional configuration `type`
* Changes of the notification topic.
* Ownership of new configuration file. 

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: docs/src/references/c8y-configuration-management.md [hunk 1]
```diff
@@ -72,20 +73,26 @@ The `c8y_configuration_plugin` configuration is stored by default under `/etc/te
...
 
 ```shell
 $ cat /etc/tedge/c8y/c8y-configuration-plugin.toml
 files = [
-    { path = '/etc/tedge/tedge.toml' },
+    { path = '/etc/tedge/tedge.toml', type = 'tedge.toml' },
     { path = '/etc/tedge/mosquitto-conf/c8y-bridge.conf' },
     { path = '/etc/tedge/mosquitto-conf/tedge-mosquitto.conf' },
-    { path = '/etc/mosquitto/mosquitto.conf' }
+    { path = '/etc/mosquitto/mosquitto.conf', type = 'mosquitto'}
```
Comment (rina23q) [inline docs/src/references/c8y-configuration-management.md:88]: I would add `user`, `group`, and `mode` example.
```suggestion
    { path = '/etc/mosquitto/mosquitto.conf', type = 'mosquitto', user = 'mosquitto', group = 'mosquitto', mode = 0o644 }
```

**Review by rina23q — APPROVED** [review]
Thanks Didier, only one minor suggestion.

File: docs/src/references/c8y-configuration-management.md (hunk 1 above)
Comment (didier-wenzek) [reply docs/src/references/c8y-configuration-management.md]: Added

File: docs/src/references/c8y-configuration-management.md [hunk 2]
```diff
@@ -72,20 +73,26 @@ The `c8y_configuration_plugin` configuration is stored by default under `/etc/te
 This [TOML](https://toml.io/en/) file defines the list of files to be managed from the cloud tenant.
 Each configuration file is defined by a record with:
 * The full `path` to the file.
+* An optional configuration `type`. If not provided, the `path` is used a `type`.
```
Comment (albinsuresh) [inline docs/src/references/c8y-configuration-management.md:76]:
```suggestion
* An optional configuration `type`. If not provided, the `path` is used as `type`.
```

File: docs/src/references/c8y-configuration-management.md [hunk 3]
```diff
@@ -144,12 +151,15 @@ The `c8y_configuration_plugin` reports progress and errors on its `stderr`.
...
-
-Points that still need to be addressed:
-
-* Ability to give a type name to a configuration file.
-* What if the target file doesn't exist? Is this is an error? If not what user, group and mod are to be used?
\ No newline at end of file
+Note that:
+* If no specific type has been assigned to a configuration file, then the path to this file is used as its type.
+  Update notifications for that file are then published on the topic `tedge/configuration_change/{path}`,
+  for instance `tedge/configuration_change//etc/tedge/mosquitto-conf/c8y-bridge.conf`.
```
Comment (albinsuresh) [inline docs/src/references/c8y-configuration-management.md:162]: That double `//` after `tedge/configuration_change` is how it should be? Shouldn't we remove that redundant `/`?

**Review by albinsuresh — APPROVED** [review]

File: docs/src/references/c8y-configuration-management.md (hunk 3 above)
Comment (didier-wenzek) [reply docs/src/references/c8y-configuration-management.md:162]: This is the topic actually used when the path is used as type.

File: docs/src/references/c8y-configuration-management.md (hunk 2 above)
Comment (didier-wenzek) [reply docs/src/references/c8y-configuration-management.md]: Fixed

File: docs/src/references/c8y-configuration-management.md (hunk 3 above)
Comment (rina23q) [reply docs/src/references/c8y-configuration-management.md:162]: If the `path` is set as `/etc/tedge/mosquitto-conf/c8y-bridge.conf` and no `type` is given, `/etc/tedge/mosquitto-conf/c8y-bridge.conf` will be treated as `type`. Therefore, the notification topic has double slashes. If the user doesn't want it, they should define the type.

**Pull request merged**